test/widgets/action_sheet_test.dart

```diff
@@ -1188,6 +1189,207 @@ void main() {
+
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [message]).toJson());
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const MessageListPage(initNarrow: CombinedFeedNarrow())));
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.descendant(
+        of: find.byType(RecipientHeader),
+        matching: find.text(message.displayRecipient ?? '')));
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    final actionSheetFinder = find.byType(BottomSheet);
+    Finder findButtonForLabel(String label) =>
+      find.descendant(of: actionSheetFinder, matching: find.text(label));
+
+    void checkButton(String label) {
+      check(findButtonForLabel(label)).findsOne();
+    }
+
+    group('showChannelActionSheet', () {
+      void checkButtons() {
+        check(actionSheetFinder).findsOne();
+        checkButton('Mark channel as read');
+      }
+
+      testWidgets('show from inbox', (tester) async {
+        await prepare();
+        await showFromInbox(tester);
+
+        checkButtons();
+      });
+
+      testWidgets('show from subscription list', (tester) async {
+        await prepare();
+        await showFromSubscriptionList(tester);
+
+        checkButtons();
+      });
+
+      testWidgets('show with no unread messages', (tester) async {
+        await prepare(hasUnreadMessages: false);
+        await showFromSubscriptionList(tester);
+
+        check(actionSheetFinder).findsNothing();
+      });
+
+      testWidgets('show from app bar in channel narrow', (tester) async {
+        await prepare();
+        final channel = someChannel;
+        final narrow = ChannelNarrow(channel.streamId);
+
+        await showFromAppBar(tester, narrow: narrow);
+
+        checkButtons();
+      });
+
+      testWidgets('show from app bar in topic narrow', (tester) async {
+        await prepare();
+        final narrow = TopicNarrow(someChannel.streamId, TopicName(someTopic));
+
+        await showFromAppBar(tester, narrow: narrow);
+
+        checkButtons();
+      });
+
+      testWidgets('show from recipient header', (tester) async {
+        await prepare();
+        final message = someMessage;
+        await showFromRecipientHeader(tester, message: message);
+
+        checkButtons();
+      });
+    });
+
+    group('MarkChannelAsReadButton', () {
+      void checkRequest(int channelId) {
+        check(connection.takeRequests()).single.isA<http.Request>()
+          ..method.equals('POST')
+          ..url.path.equals('/api/v1/messages/flags/narrow')
+          ..bodyFields.deepEquals({
+            'anchor': 'oldest',
+            'include_anchor': 'false',
+            'num_before': '0',
+            'num_after': '1000',
+            'narrow': jsonEncode([
+              {'operator': 'stream', 'operand': channelId},
+              {'operator': 'is', 'operand': 'unread'}
+            ]),
```

Comment:
nit: always use trailing comma unless the closing delimiter (`]`, etc.) is on the same line:
```suggestion
              {'operator': 'is', 'operand': 'unread'},
            ]),
```